### DISCUSSION TITLE
Ideal way to store user passwords in Supabase

### DISCUSSION THREAD

**Discussion by rdong8** [sqlalchemy/sqlalchemy#12185] [Usage Questions] [answered]
What would be the best way to store password-like data in Supabase? Specifically I have a user data table indexed by integer user IDs, I'd like to associate some password-like data to each of these users. Would the encryption at rest be enough to just store the password in a string column of this table? Or should I use vault?

**Comment by CaselIT** [comment, marked as answer]
Hi,

This looks like a question for the supabase support forum not for sqlalchemy one.

A general advice is use not store the password, but to use one of hash function recommended for password, like argon2, and store the hash of the password in the db instead.
This is a very common case so you can find a lot of resources on this topic
